### PR TITLE
Add dhall lsp server

### PR DESCRIPTION
**Pull request by oberblastmeister**

### CODE REVIEW

**Comment by mjlbach** [comment]
Can you mention the installation? It's installable via cabal

dhall-lsp-server can be installed be installed via cabal: 

```bash
cabal install dhall-lsp-server
```

**Comment by mjlbach** [comment]
the 'via x' wording is kinda standard across different servers/npm which is why I suggested it

**Comment by mjlbach** [comment]
Also I would still say `dhall-lsp-server` can be installed since that is the package name (sorry for the nitpicking), I'll merge this after you do it

**Comment by mjlbach** [comment]
I don't think this line is necessary `Make sure to pick the one that says dhall-lsp-server-*`

**Comment by oberblastmeister** [comment]
> I would still say dhall-lsp-server can be installed since that is the package name

Sorry, what do you mean by that?

File: lua/lspconfig/dhall_lsp_server.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+https://github.com/dhall-lang/dhall-haskell/tree/master/dhall-lsp-server
+
+language server for dhall
+
+`dhall_lsp_server` can be installed via cabal:
+```sh
+cabal install dhall-lsp-server
+```
+prebuilt binaries can be found [here](https://github.com/dhall-lang/dhall-haskell/releases).
+Make sure to pick the one that says dhall-lsp-server-*
```
Comment (mjlbach) [inline lua/lspconfig/dhall_lsp_server.lua:20]: Delete this line

**Comment by mjlbach** [comment]
Great! Squash and we can merge

**Comment by oberblastmeister** [comment]
Sorry for being slow. I am not so good at git yet, I learned programming like a year ago. This is my first time squashing so I had to search it up. Not sure if I did it correctly.

**Comment by mjlbach** [comment]
```
git remote add upstream git@github.com:neovim/nvim-lspconfig.git
git fetch upstream
git rebase -i upstream/master
```
Now change all but the first pick to squash, and change the first pick to reword.

Then change the commit message (it should pop up) to `added dhall-lsp-server` and hit :wq

then run `git diff upstream/master` to make sure that the changes reflect what you want, and then `git push -f ` to overwrite your commit history for the branch. You should have only a single commit that reflects all of this work then (which keeps our git history clean).

**Comment by oberblastmeister** [comment]
did I do it correctly?

**Comment by mjlbach** [comment]
Yes! (interactive rebase is the best)

**Comment by oberblastmeister** [comment]
:smile:

**Comment by oberblastmeister** [comment]
thanks for helping me!

**Pull request merged**